eventstore/models.py

```diff
@@ -1094,3 +1094,39 @@ def clean(self):
                 date(self.dob_year, self.dob_month, self.dob_day)
             except ValueError as e:
                 raise ValidationError(f"Invalid date of birth date, {str(e)}")
+
+
+class OpenHIMQueue(models.Model):
+    class ObjectType:
+        PREBIRTH_REGISTRATION = "PrebirthRegistration"
+        CHW_REGISTRATION = "CHWRegistration"
+        PUBLIC_REGISTRATION = "PublicRegistration"
+        CHANNEL_SWITCH = "ChannelSwitch"
+        OPTOUT = "OptOut"
+        choices = (
+            (PREBIRTH_REGISTRATION, "PrebirthRegistration"),
+            (CHW_REGISTRATION, "CHWRegistration"),
+            (PUBLIC_REGISTRATION, "PublicRegistration"),
+            (CHANNEL_SWITCH, "ChannelSwitch"),
+            (OPTOUT, "OptOut"),
+        )
+
+    class Status:
+        PENDING = 0
+        PROCESSING = 1
+        COMPLETE = 2
+        ERROR = 3
+
+        choices = (
+            (PENDING, "Pending"),
+            (PROCESSING, "Processing"),
+            (COMPLETE, "Complete"),
+            (ERROR, "Error"),
+        )
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    object_id = models.UUIDField()
+    object_type = models.CharField(max_length=30, choices=ObjectType.choices)
+    status = models.PositiveSmallIntegerField(
+        choices=Status.choices, default=Status.PENDING
+    )
```

Comment:
Do we also need to store a copy of the details from the object? In case they do a forget delete before this gets forwarded? Or in that case do we not want to forward the object to them?

Reply:
Hmm I didn't think about that use case, I would assume we don't send it then, since they requested we delete all their data. We can't leave it in a queue then

Reply:
I've noted the assumption on the ticket